templates/commands/render/action_include.go

```diff
@@ -44,9 +42,111 @@ func actionInclude(ctx context.Context, inc *spec.Include, sp *stepParams) error
 	return nil
 }
 
-func includePath(ctx context.Context, inc *spec.IncludePath, sp *stepParams) error {
+func createSkipMap(ctx context.Context, inc *spec.IncludePath, sp *stepParams, fromDir string) (map[string]struct{}, error) {
+	skip := make(map[string]struct{}, len(inc.Skip))
+
+	unglobbedSkipPaths, err := processPaths(inc.Skip, sp.scope)
+	if err != nil {
+		return nil, err
+	}
+	skipPaths, err := processGlobs(ctx, unglobbedSkipPaths, fromDir, sp.features.SkipGlobs)
```

Comment:
I think this might be less efficient than it could be. This code as written expans the skip glob into a list of files, and then checks that list for each file that gets crawled by copyParams. The first step isn't actually necessary: we can check each filename against each glob using filepath.Match(). That's the approach used at https://github.com/abcxyz/abc/blob/12290a2cf560efcb2c5b3aad606d9de1a08f0240/templates/commands/render/action_include.go#L164.

Reply:
Isn't the current approach more efficient? Imagine we have S number of skip globs and P number of paths.

If we expand all of the skip globs and put them into the map then check if each path is in the map as they're processed (current behavior), I think this is S + P operations.

If we don't expand the skip globs but keep them as a list, for each path we encounter, we must check filepath.Match for all of the skip globs to see if it needs to be skipped. I think this is S * P operations?

Reply:
Yeah, I see what you mean. There is some unpleasantness in doing S*P match operations. But those are almost free compared to the cost of filesystem operations. We can avoid S calls to filepath.Glob(), which is a system call that does IO, which will be far more expensive than an in-memory Match() operation.

Reply:
Ok, that sounds good to me! I didn't realize that filepath.Glob was so expensive.

Reply:
Use filepath.Match, removed skip processGlobs.